### PR TITLE
[Core/AWS] Skip association with no subnetId and fix autodown for spot instances

### LINKED ISSUE #2920: [AWS] KeyError 'SubnetId' when multiple Main route tables appear in a same region
`sky launch --cloud aws --region us-west-2 --cpus 2 echo hi`
```
E 12-29 15:55:13 provisioner.py:88] Failed to configure the cloud for 'test-west' with the following error:
E 12-29 15:55:13 provisioner.py:88] KeyError: 'SubnetId'
W 12-29 15:55:14 cloud_vm_ray_backend.py:2246] sky.exceptions.ResourcesUnavailableError: Failed to acquire resources in all zones in us-west-2. Try changing resource requirements or use another region.
```

It is mainly because the route table associations can have both main:true and main:false in the same route table.
`aws ec2 describe-route-tables --region us-west-2`
```
{
    "RouteTables": [
        {
            "Associations": [
                {
                    "Main": false,
                    "RouteTableAssociationId": "rtbassoc-xxxxxx",
                    "RouteTableId": "rtb-xxxxxx",
                    "SubnetId": "subnet-xxxxxx",
                    "AssociationState": {
                        "State": "associated"
                    }
                },
                {
                    "Main": false,
                    "RouteTableAssociationId": "rtbassoc-xxxxxx",
                    "RouteTableId": "rtb-xxxxxx",
                    "SubnetId": "subnet-xxxxxx",
                    "AssociationState": {
                        "State": "associated"
                    }
                },
                {
                    "Main": false,
                    "RouteTableAssociationId": "rtbassoc-xxxxxx",
                    "RouteTableId": "rtb-xxxxxx",
                    "SubnetId": "subnet-xxxxxx",
                    "AssociationState": {
                        "State": "associated"
                    }
                },
                {
                    "Main": false,
                    "RouteTableAssociationId": "rtbassoc-xxxxxx",
                    "RouteTableId": "rtb-xxxxxx",
                    "SubnetId": "subnet-xxxxxx",
                    "AssociationState": {
                        "State": "associated"
                    }
                },
                {
                    "Main": true,
                    "RouteTableAssociationId": "rtbassoc-xxxxxx",
                    "RouteTableId": "rtb-xxxxxx",
                    "AssociationState": {
                        "State": "associated"
                    }
                }
            ],
            "PropagatingVgws": [],
            "RouteTableId": "rtb-xxxxxx",
            "Routes": [
                {
                    "DestinationCidrBlock": "10.0.0.0/16",
                    "GatewayId": "local",
                    "Origin": "CreateRouteTable",
                    "State": "active"
                },
                {
                    "DestinationCidrBlock": "0.0.0.0/0",
                    "GatewayId": "igw-xxxxxx",
                    "Origin": "CreateRoute",
                    "State": "active"
                }
            ],
            "Tags": [],
            "VpcId": "vpc-xxxxxx",
            "OwnerId": "xxxxxxxx"
        },
        {
            "Associations": [
                {
                    "Main": true,
                    "RouteTableAssociationId": "rtbassoc-xxxxxx",
                    "RouteTableId": "rtb-xxxxxx",
                    "AssociationState": {
                        "State": "associated"
                    }
                }
            ],
            "PropagatingVgws": [],
            "RouteTableId": "rtb-xxxxxx",
            "Routes": [
                {
                    "DestinationCidrBlock": "172.31.0.0/16",
                    "GatewayId": "local",
                    "Origin": "CreateRouteTable",
                    "State": "active"
                },
                {
                    "DestinationCidrBlock": "0.0.0.0/0",
                    "GatewayId": "igw-xxxxxx",
                    "Origin": "CreateRoute",
                    "State": "active"
                }
            ],
            "Tags": [],
            "VpcId": "vpc-xxxxxx",
            "OwnerId": "xxxxxx"
        }
    ]
}
```

### PR DESCRIPTION
**Pull request by Michaelvll**
Fixes #2920.

Tested (run the relevant ones):

- [ ] Code formatting: `bash format.sh`
- [ ] Any manual or new tests for this PR (please specify below)
  - [x] `sky launch --cloud aws --region us-west-2 -c test-west --cpus 2 echo hi`
- [ ] All smoke tests: `pytest tests/test_smoke.py` 
- [ ] Relevant individual smoke tests: `pytest tests/test_smoke.py::test_fill_in_the_name` 
- [ ] Backward compatibility tests: `bash tests/backward_comaptibility_tests.sh`

### CODE REVIEW

File: sky/provision/aws/config.py [hunk 1]
```diff
@@ -218,7 +218,10 @@ def _is_subnet_public(ec2, subnet_id, vpc_id: Optional[str]) -> bool:
         rt for rt in all_route_tables
         # An RT can be associated with multiple subnets, i.e.,
         # rt['Associations'] is a list of associations.
-        if any(assoc['SubnetId'] == subnet_id for assoc in rt['Associations'])
+        # There may be no subnet associated in a association, we skip it.s
```
Comment (concretevitamin) [inline sky/provision/aws/config.py:221]:
```suggestion
        # There may be no subnet associated in an association.
```

**Review by concretevitamin — APPROVED** [review]
LGTM.

Btw,
```
sky launch --cloud aws -i0 --down --use-spot -y --region us-east-2 
```
launched successfully but ran into
```

/Users/zongheng/anaconda/envs/py38/lib/python3.8/multiprocessing/resource_tracker.py:216: UserWarning: resource_tracker: There appear to be 4 leaked semaphore objects to clean up at shutdown
  warnings.warn('resource_tracker: There appear to be %d '
sky.exceptions.NotSupportedError: The following features are not supported by AWS:
        Feature  Reason
        stop     Stopping spot instances is currently not supported on AWS.
```
which appears to be a bug in the last PR (we probably should check `--down`).

**Comment by Michaelvll** [comment]
> LGTM.
> 
> Btw,
> 
> ```
> sky launch --cloud aws -i0 --down --use-spot -y --region us-east-2 
> ```
> 
> launched successfully but ran into
> 
> ```
> 
> /Users/zongheng/anaconda/envs/py38/lib/python3.8/multiprocessing/resource_tracker.py:216: UserWarning: resource_tracker: There appear to be 4 leaked semaphore objects to clean up at shutdown
>   warnings.warn('resource_tracker: There appear to be %d '
> sky.exceptions.NotSupportedError: The following features are not supported by AWS:
>         Feature  Reason
>         stop     Stopping spot instances is currently not supported on AWS.
> ```
> 
> which appears to be a bug in the last PR (we probably should check `--down`).

Great catch! Just fixed it in this PR. PTAL.

**Review by concretevitamin — APPROVED** [review]
LGTM, thanks @Michaelvll.

**Comment by concretevitamin** [comment]
With 
```
sky launch --cloud aws -i0 --down --use-spot -y --region us-east-2 
```
it now shows the following at the end
```
...

* 1 cluster has auto{stop,down} scheduled. Refresh statuses with: sky status --refresh

/Users/zongheng/anaconda/envs/py38/lib/python3.8/multiprocessing/resource_tracker.py:216: UserWarning: resource_tracker: There appear to be 4 leaked semaphore objects to clean up at shutdown
```

**Pull request merged**